diff -Nru boxes-1.0.1.orig/src/lexer.l boxes-1.0.1/src/lexer.l
--- boxes-1.0.1.orig/src/lexer.l	2000-03-18 00:53:59.000000000 +0100
+++ boxes-1.0.1/src/lexer.l	2006-06-15 02:47:12.507645100 +0200
@@ -116,7 +116,6 @@
     "$Id: lexer.l,v 1.17 1999/08/20 19:51:12 tsjensen Exp tsjensen $";
 
 
-int yylineno = 1;
 static int yyerrcnt = 0;
 
 static char sdel = '\"';
@@ -138,6 +137,7 @@
 %option noyywrap
 %option never-interactive
 %option caseless
+%option noyylineno
 
 
 %x SAMPLE
@@ -468,7 +468,7 @@
         exit (EXIT_FAILURE);
     }
     yy_delete_buffer (YY_CURRENT_BUFFER);
-    YY_CURRENT_BUFFER = yy_create_buffer (yyin, sinf.st_size+10);
+    yy_switch_to_buffer (yy_create_buffer (yyin, sinf.st_size+10));
 }
 
 
diff -Nru boxes-1.0.1.orig/src/regexp/Makefile boxes-1.0.1/src/regexp/Makefile
--- boxes-1.0.1.orig/src/regexp/Makefile	2000-03-18 00:54:00.000000000 +0100
+++ boxes-1.0.1/src/regexp/Makefile	2006-06-15 02:46:04.877614708 +0200
@@ -32,7 +32,7 @@
 #============================================================================
 
 
-CFLAGS   = -traditional -O -I. $(CFLAGS_ADDTL)
+CFLAGS   = -O -I. $(CFLAGS_ADDTL)
 
 ALL_CL   = regexp/regexp.c regexp/regsub.c
 C_SRC    = $(notdir $(ALL_CL))
diff -Nru boxes-1.0.1.orig/src/regexp/regexp.c boxes-1.0.1/src/regexp/regexp.c
--- boxes-1.0.1.orig/src/regexp/regexp.c	2000-03-18 00:54:00.000000000 +0100
+++ boxes-1.0.1/src/regexp/regexp.c	2006-06-15 02:49:18.027413460 +0200
@@ -23,6 +23,8 @@
  * regular-expression syntax might require a total rethink.
  */
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <regexp.h>
 #include "regmagic.h"
 
@@ -197,7 +199,6 @@
 	register char *longest;
 	register int len;
 	int flags;
-	extern char *malloc();
 
 	if (exp == NULL)
 		FAIL("NULL argument");
@@ -706,7 +707,6 @@
 register char *string;
 {
 	register char *s;
-	extern char *strchr();
 
 	/* Be paranoid... */
 	if (prog == NULL || string == NULL) {
@@ -805,7 +805,6 @@
 {
 	register char *scan;	/* Current node. */
 	char *next;		/* Next node. */
-	extern char *strchr();
 
 	scan = prog;
 #ifdef DEBUG
diff -Nru boxes-1.0.1.orig/src/regexp/regsub.c boxes-1.0.1/src/regexp/regsub.c
--- boxes-1.0.1.orig/src/regexp/regsub.c	2000-03-18 00:54:00.000000000 +0100
+++ boxes-1.0.1/src/regexp/regsub.c	2006-06-15 02:50:13.361074690 +0200
@@ -42,6 +42,7 @@
  */
 
 #include <stdio.h>
+#include <string.h>
 #include <regexp.h>
 #include "regmagic.h"
 
@@ -74,7 +75,6 @@
     register int no;
     register int len;
     size_t fill;                         /* current number of chars in dest */
-    extern char *strncpy();
 
     if (prog == NULL || source == NULL || dest == NULL) {
         regerror("NULL parm to regsub");
diff -Nru boxes-1.0.1a.orig/Makefile boxes-1.0.1a/Makefile
--- boxes-1.0.1a.orig/Makefile	2000-03-18 00:53:58.000000000 +0100
+++ boxes-1.0.1a/Makefile	2006-06-18 17:48:28.000000000 +0200
@@ -51,8 +51,8 @@
 
 
 # The following line (GLOBALCONF) is the only line you should need to edit!
-GLOBALCONF = /usr/local/share/boxes
-BVERSION   = 1.0.1
+GLOBALCONF = /etc/boxes/boxes-config
+BVERSION   = 1.0.1a
 
 SNAPFILE   = boxes-SNAP-$(shell date +%Y%m%d)
 WEBHOME    = $(HOME)/d/public_html/software/boxes
